src/new_index/db_metrics.rs

```diff
@@ -0,0 +1,225 @@
+                "rocksdb_mem_table_flush_pending",
+                "Number of pending memtable flushes"
+            ), labels),
+            cur_size_active_mem_table: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_cur_size_active_mem_table_bytes",
+                "Current size of active memtable in bytes"
+            ), labels),
+            cur_size_all_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_cur_size_all_mem_tables_bytes",
+                "Current size of all memtables in bytes"
+            ), labels),
+            size_all_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_size_all_mem_tables_bytes",
+                "Size of all memtables in bytes"
+            ), labels),
+            num_entries_active_mem_table: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_num_entries_active_mem_table",
+                "Number of entries in active memtable"
+            ), labels),
+            num_entries_imm_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                "rocksdb_num_entries_imm_mem_tables",
+                "Number of entries in immutable memtables"
+            ), labels),
+            num_deletes_active_mem_table: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_deletes_active_mem_table"),
+                "Number of delete entries in active memtable"
+            ), labels),
+            num_deletes_imm_mem_tables: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_deletes_imm_mem_tables"),
+                "Number of delete entries in immutable memtables"
+            ), labels),
+
+            // Compaction metrics
+            compaction_pending: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_compaction_pending"),
+                "Number of pending compactions"
+            ), labels),
+            estimate_pending_compaction_bytes: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_pending_compaction_bytes"),
+                "Estimated bytes pending compaction"
+            ), labels),
+            num_running_compactions: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_running_compactions"),
+                "Number of currently running compactions"
+            ), labels),
+            num_running_flushes: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_num_running_flushes"),
+                "Number of currently running flushes"
+            ), labels),
+
+            // Error metrics
+            background_errors: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_background_errors_total"),
+                "Total number of background errors"
+            ), labels),
+
+            // Key and data size estimates
+            estimate_num_keys: metrics.gauge_vec(MetricOpts::new(
+                format!("rocksdb_estimate_num_keys"),
+                "Estimated number of keys"
```

Comment:
estimated number of total keys in the active and unflushed immutable memtables and storage